### ISSUE TITLE
Always prompts for kernel selection

### ISSUE THREAD

**Issue by ryanlovett** [jupyterlab/jupyterlab#14404]
## Description

When I ask lab to open a notebook, it always prompts me to select a kernel. The kernel is already saved in the .ipynb metadata and the kernel is installed. Given the file `lec02.ipynb`:

```bash
(notebook) jovyan@jupyter-rylo:~/sp23/lecture/lec02$ jq .metadata.kernelspec lec02.ipynb
{
  "display_name": "Python 3 (ipykernel)",
  "language": "python",
  "name": "python3"
}
(notebook) jovyan@jupyter-rylo:~/sp23/lecture/lec02$ jupyter kernelspec list
Available kernels:
  bash       /srv/conda/envs/notebook/share/jupyter/kernels/bash
  python3    /srv/conda/envs/notebook/share/jupyter/kernels/python3
(notebook) jovyan@jupyter-rylo:~/sp23/lecture/lec02$
```

This also happens with new notebook created by lab.

## Reproduce

1. Navigate to .ipynb file.
2. Double-click on it.
3. Lab prompts for the kernel choice. See screenshot:

<img width="320" alt="Screenshot 2023-04-18 at 1 43 09 PM" src="https://user-images.githubusercontent.com/10248125/232900402-e27acb12-a83c-4b47-bfeb-b257d5d23309.png">

## Expected behavior

I expected lab to automatically choose the only available kernel in the kernelspec that matches the metadata in the notebook, without prompting me. I opened classic notebook in the same environment and it was able to open the .ipynb file without prompting for kernel choice.
## Context

- Operating System and version: Ubuntu 18.04 (repo2docker)
- Browser and version: Chrome 108
- JupyterLab version: 3.6.3

The environment is built using repo2docker@968cc43 and is defined at https://github.com/berkeley-dsep-infra/datahub/tree/staging/deployments/stat159/image.

**Comment by ryanlovett** [comment]
(i had to split the issue into two parts because github told me it was longer than 65536 bytes.)

**Comment by ryanlovett** [comment]
It is not reproducible in Notebook within the same environment, only JupyterLab.

I pulled and ran the same docker image outside of jupyterhub. I _could not_ reproduce when run simply as as:

`docker run --rm -p 8888:8888 us-central1-docker.pkg.dev/ucb-datahub-2018/user-images/stat159-user-image:07efb6f jupyter lab --ip=0.0.0.0`

I _could_ reproduce when run as either:

`docker run --rm -p 8888:8888 us-central1-docker.pkg.dev/ucb-datahub-2018/user-images/stat159-user-image:07efb6f jupyter lab --ip=0.0.0.0 --LabApp.collaborative=true`

`docker run --rm -p 8888:8888 us-central1-docker.pkg.dev/ucb-datahub-2018/user-images/stat159-user-image:07efb6f jupyter lab --ip=0.0.0.0 --LabApp.collaborative=true --YDocExtension.ystore_class=tmpystore.TmpYStore`

We enable RTC in this particular jupyterhub so we do use the last two args there.

I'll build a stripped down version of this image so it is easier to reproduce.

**Comment by ryanlovett** [comment]
I'm able to reproduce with the conda `environment.yml` file:

```yaml
name: stat159-s23

channels:
  - conda-forge

dependencies:
  - python==3.10.8
  - jupyter_server==2.5.0
  - jupyterlab==3.6.3
  - jupyterlab_server==2.22.1
  - jupyter_server_ydoc==0.8.0
```

and building the image with `repo2docker --no-run .` I ran the image with

```bash
docker run --rm -p 8888:8888 r2d-{image_hash}:latest jupyter lab --ip=0.0.0.0 --LabApp.collaborative=true
```

When lab started, I made a new notebook with the python kernel, evaluated something in a cell, then shut down the kernel and closed the notebook. When I reopened it, lab prompted me to choose a kernel.

**Comment by davidbrochart** [comment]
I can reproduce this issue with JupyterLab 4.0.0b2 and `jupyter_collaboration` installed.

**Comment by yzhum** [comment]
Is there a plan to backport the fix to 3.6.x?
